src/main/java/org/example/api/DeliveryEmployeeService.java

```diff
@@ -41,7 +45,12 @@ public int createDeliveryEmployee(DeliveryEmployeeRequest deliveryEmployeeReques
      * @throws DoesNotExistException if no employee returned from database
      * @throws FailedToGetException if sql statement failed
      */
-    public DeliveryEmployee getDeliveryEmployeeById(int id) throws DoesNotExistException, FailedToGetException {
+    public DeliveryEmployee getDeliveryEmployeeById(int id) throws DoesNotExistException, FailedToGetException, DeliveryEmployeeDoesNotExistException {
+
+        // call to dao to check employee with given id is a delivery employee
+        if(dao.checkEmployeeIsDeliveryEmployee(id) == null){
+            throw new DeliveryEmployeeDoesNotExistException();
+        }
```

Comment:
could you not add the null check after the call below which gets the delivery employee?